File: packages/react-native-storybook/android/src/main/java/io/sherlo/storybookreactnative/SherloJsonHelper.java
package io.sherlo.storybookreactnative;

import android.content.Context;
import android.util.Log;
import org.json.JSONObject;

/**
 * Helper class for reading sherlo.json from bundled assets.
 */
public class SherloJsonHelper {
    private static final String TAG = "SherloModule:SherloJsonHelper";

    /**
     * Returns the native version string from sherlo.json bundled assets, or null if missing/unreadable.
     *
     * @param context The Android context
     * @return The version string, or null if missing or unreadable
     */
    public static String getNativeVersion(Context context) {
        try {
            java.io.InputStream is = context.getAssets().open("sherlo.json");
            java.io.ByteArrayOutputStream baos = new java.io.ByteArrayOutputStream();
            byte[] buf = new byte[1024];
            int len;
            while ((len = is.read(buf)) != -1) baos.write(buf, 0, len);
            JSONObject sherloJson = new JSONObject(baos.toString("UTF-8"));
            return sherloJson.optString("version", null);
        } catch (Exception e) {
            Log.w(TAG, "Could not read sherlo.json: " + e.getMessage());
            return null;
        }
    }
}

File: packages/react-native-storybook/ios/SherloJsonHelper.m
#import "SherloJsonHelper.h"

static NSString *const LOG_TAG = @"SherloModule:SherloJsonHelper";

/**
 * Helper for reading sherlo.json from bundled assets.
 */
@implementation SherloJsonHelper

/**
 * Returns the native version string from sherlo.json bundled assets, or nil if missing/unreadable.
 *
 * @return The version string, or nil if missing or unreadable
 */
+ (NSString *)getNativeVersion {
    NSBundle *podBundle = [NSBundle bundleForClass:[SherloJsonHelper class]];
    NSString *sherloJsonPath = [podBundle pathForResource:@"sherlo" ofType:@"json" inDirectory:@"assets"];
    if (!sherloJsonPath) {
        return nil;
    }
    NSData *sherloJsonData = [NSData dataWithContentsOfFile:sherloJsonPath];
    if (!sherloJsonData) {
        return nil;
    }
    NSDictionary *sherloJson = [NSJSONSerialization JSONObjectWithData:sherloJsonData options:0 error:nil];
    return sherloJson[@"version"];
}

@end
